src/servers/grpc_server_v2.cc

```diff
@@ -840,12 +841,16 @@ CommonHandler::SetUpAllRequests()
         TRTSERVER_Error* err = nullptr;
 #ifdef TRTIS_ENABLE_GPU
         const std::string& raw_handle = request.raw_handle();
-        char* handle_base = const_cast<char*>(raw_handle.c_str());
-        cudaIpcMemHandle_t* cuda_shm_handle =
-            reinterpret_cast<cudaIpcMemHandle_t*>(handle_base);
+        // Decode the handle from base64
+        void* cuda_shm_handle = malloc(raw_handle.size());
```

Comment:
There are a few `malloc()` calls for encoding / decoding purpose, are the resources being cleaned up properly?